teps/0001-tekton-enhancement-proposal-process.md

```diff
@@ -0,0 +1,473 @@
+
+1. An initial design doc is crafted (let's imagine it is [Tekton OCI
+Image
+Catalog](https://docs.google.com/document/d/1zUVrIbGZh2R9dawKQ9Hm1Cx3GevKIfOcRO3fFLdmBDc/edit#heading=h.tp9mko2koenr)).
+   An experimental project has already been created and a
+   proof-of-concept demoed during a working group. The next step is to
+   create a TEP (and continue work on the proof-of-concept if need be).
+2. From this design docs, a TEP is created with the content of the
+   design document.
+3. It is approved with a `proposed` state, which means :
+   - We acknowledge this is important for the project, and needs to be
+     worked on
+   - It needs some work and discussion based on the initial proposal
+4. The TEP is being disscussed during Working Group(s) — it can be the
+   main one, a specific one, the API Working Group.
+
+   During those discussion it is clear that some work needs to be
+   done:
+   - Define a Spec for the OCI image (layers, metadata, configuration)
+     The experimental project can be used to demo and validate that spec.
+   - Once the spec is agreed on
+   - Have a new TEP to add support for referencing Task and Pipeline
+     through alternative means than in clusters (OCI image is one,
+     using Git or an HTTP url are others)
+
+   The next actions are :
+   - Create a new TEP on support for referencing Task and Pipeline.
+     As before, the TEP can be first discussed during Working group
+     and refined in Google Docs before being proposed as a TEP.
+   - Update the current TEP to define the spec (same thing as above
+     applies). A name is choosed for those: Tekton Bundles.
+   - Create a new TEP on implementing Tekton Bundles in tektoncd
+     projects (`pipeline` and `cli`)
+5. The current TEP, defining the spec, is *approved* and marked as
+   `implemented`. In this case `implemented` means it is available in
+   the documentation in `tektoncd` (most likely on the
+   `tektoncd/pipeline` repository)
+
+We are now switching to the "Implementing Tekton Bundles" TEP.
+
+1. It is proposed based on a design docs (discussed during working
+   group)
+2. It depends on the TEP on support for referencing Task and Pipeline
+   to be approved and in the `implementable` (aka we have agreed on
+   how it should be implemented more or less 😝).
+   It is also agreed (and most likely written in the TEP) that
+   "Implementing Tekton Bundle" would serve as the first
+   implementation of this TEP.
+3. The "Implementing Tekton Bundle" gets approved, and as it has been
+   discussed during working groups, it is ready for implementation, so
+   it gets merged directly into `implementable`.
+4. Work is happening in `tektoncd/pipeline` (and `tektoncd/cli` in
+   parallel) on implementing it.
+5. Implementation is done, we update the TEP to put it in
+   `implemented` state.
+
+### PipelineResource re-design
+
+For more context, this is linked to the PipelineResource work and more
+accurately the following:
```

Comment:
im not sure what this sentence is getting at, similar to the previous section maybe this could be something like "here are some links for context"?